staging/src/github.com/openshift/sdn/pkg/openshift-sdn/cmd.go

```diff
@@ -123,47 +128,38 @@ func (sdn *OpenShiftSDN) Run(c *cobra.Command, errout io.Writer, stopCh chan str
 // ValidateAndParse validates the command line options, parses the node
 // configuration, and builds the upstream proxy configuration.
 func (sdn *OpenShiftSDN) ValidateAndParse() error {
-	if len(sdn.ConfigFilePath) == 0 {
-		return errors.New("--config is required")
-	}
-
-	if len(sdn.KubeConfigFilePath) > 0 && len(sdn.URLOnlyKubeConfigFilePath) > 0 {
-		return errors.New("cannot pass --kubeconfig and --url-only-kubeconfig")
-	}
-
-	klog.V(2).Infof("Reading node configuration from %s", sdn.ConfigFilePath)
-	var err error
-	sdn.NodeConfig, err = readAndResolveNodeConfig(sdn.ConfigFilePath)
-	if err != nil {
-		return err
-	}
-
-	if len(sdn.KubeConfigFilePath) > 0 {
-		sdn.NodeConfig.MasterKubeConfig = sdn.KubeConfigFilePath
-	}
+	sdn.nodeName = os.Getenv("K8S_NODE_NAME")
 
-	// Get the nodename from the environment, if available
-	if len(sdn.NodeConfig.NodeName) == 0 {
-		sdn.NodeConfig.NodeName = os.Getenv("K8S_NODE_NAME")
+	if len(sdn.ConfigFilePath) == 0 && len(sdn.ProxyConfigFilePath) == 0 {
+		return errors.New("Either --config or --proxy-config is required")
 	}
 
-	// Validate the node config
-	validationResults := networkconfig.ValidateInClusterNetworkNodeConfig(sdn.NodeConfig, nil)
-
-	if len(validationResults.Warnings) != 0 {
-		for _, warning := range validationResults.Warnings {
-			klog.Warningf("Warning: %v, node start will continue.", warning)
+	if sdn.ProxyConfigFilePath != "" {
+		klog.V(2).Infof("Reading proxy configuration from %s", sdn.ProxyConfigFilePath)
+		var err error
+		sdn.ProxyConfig, err = readProxyConfig(sdn.ProxyConfigFilePath)
+		if err != nil {
+			return err
 		}
-	}
-	if len(validationResults.Errors) != 0 {
-		klog.V(4).Infof("Configuration is invalid: %#v", sdn.NodeConfig)
-		return kerrors.NewInvalid(schema.GroupKind{Group: "", Kind: "NodeConfig"}, sdn.ConfigFilePath, validationResults.Errors)
-	}
-
-	sdn.ProxyConfig, err = ProxyConfigFromNodeConfig(*sdn.NodeConfig)
-	if err != nil {
-		klog.V(4).Infof("Unable to build proxy config: %v", err)
-		return err
+		sdn.ProxyConfig.HostnameOverride = sdn.nodeName
+		// FIXME: make this configurable
+		sdn.enableUnidling = true
+	} else {
+		klog.V(2).Infof("Reading proxy configuration from %s", sdn.ConfigFilePath)
```

Comment:
nit: "reading node configuration", but honestly this will be gone in a few days.